app/Http/Controllers/Api/CategoryController.php

```
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Models\Category;
+use Illuminate\Http\Request;
+
+
+class CategoryController extends Controller
+{
+    public function index()
+    {
+        $categories = Category::all();
+        $data = [
+            'data' => $categories
+        ];
+        return response()->json($data, 200);
+    }
+
+    public function store(Request $request)
+    {
+        $validated = $request->validate([
+            'name' => 'string'
+        ]);
+        try {
+            $category = new Category();
+            $category->name = $validated['name'];
+            $category->save();
+            return response()->json(['data' => $category], 201);
+
+        } catch (\Exception $e) {
+            return response()->json(['message' => 'Error al crear la categoria'], 500);
+        }
+    }
+
+    public function show($id)
+    {
+        $category = Category::find($id);
+        if (!$category) {
+            $data = [
+                'message' => 'Estudiante no encontrado'
+            ];
+            return response()->json($data, 404);
```

throw a ModelNotFound error instead of return the response